Readme.md

```diff
@@ -1,5 +1,298 @@
 # Flux
 
-lexer generator for any context free syntax in 100% rust
+A lexer generator for any context-free syntax in 100% rust with no dependencies.
 
-created for [Fender-lang](https://github.com/FenderLang/)
+Created for [Fender-lang](https://github.com/FenderLang/).
+
+# Lexer
+
+## What is a Lexer
+- A Lexer is part of an interpreter that takes sequence of characters and turns them into tokens.
+
+## How to use the Lexer
+
+First define your BNF input
+```rust
+let bnf_input = include_str!("path/to/example.bnf");
+```
+
+To parse the input you should use `flux_bnf::bnf::parse`
+```rust
+let mut lexer = flux_bnf::bnf::parse(bnf_input).unwrap();     
+```
+
+## How to set Lexer Rules
+Now that you have `Lexer` we can use that to now set some rules
+
+Before being able to set some rules you will need write a BNF describing the syntax which you can find [here](#bnf)
+
+Here is an example of what our BNF looks like
+```
+root ::= NumberList
+sep ::= " "
+number ::= [0-9]+
+NumberList ::= number (sep number)*
+```
+
+Token input using BNF
+`1 2 3`
+
+Token output
+```
+Some(
+    Token {
+        name: Some(
+            "NumberList",
+        ),
+        match: "1 2 3",
+        range: 0..5,
+        children: [
+            Token {
+                name: Some(
+                    "number",
+                ),
+                match: "1",
+                range: 0..1,
+            },
+            Token {
+                name: Some(
+                    "number",
+                ),
+                match: "2",
+                range: 2..3,
+            },
+            Token {
+                name: Some(
+                    "number",
+                ),
+                match: "3",
+                range: 4..5,
+            },
+        ],
+        ..
+    },
+)
+```
+To apply the rules to the `lexer` you simpily do
```

Comment:
simplily -> simply

Should clarify what kind of rules you mean and what they're for. It looks like you have 3 token outputs here, but I was only expecting to see 2. This first one seems to already be the pruned one without `sep` tokens. We should probably remove the first output here and expand the explanation of what the rules are for.